From d8e70e077d0269d003735b05fcb6bab2c3509e5b Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Wed, 5 Feb 2025 15:45:11 +0100
Subject: [PATCH 1601/1695] plat-stm32mp1: drop exti references from PWR IRQ

With drivers and DT using the hierarchical stack, drop the unused
calls to EXTI API.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 588bec4d18268f93c10ce1e932d6b429a1c5ace9 plat-stm32mp1: drop exti references from PWR IRQ]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I153e4b8da45f73a11a02d896850fc257a1958037
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434943
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../plat-stm32mp1/drivers/stm32mp1_pwr_irq.c  | 26 +------------------
 1 file changed, 1 insertion(+), 25 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr_irq.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr_irq.c
index 0d49758fa..5aa73bdd7 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr_irq.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pwr_irq.c
@@ -6,7 +6,6 @@
 #include <config.h>
 #include <drivers/gpio.h>
 #include <drivers/regulator.h>
-#include <drivers/stm32_exti.h>
 #include <drivers/stm32_gpio.h>
 #include <drivers/stm32mp1_pwr.h>
 #include <dt-bindings/interrupt-controller/irq.h>
@@ -47,9 +46,6 @@ enum wkup_pull_setting {
 	WKUP_PULL_RESERVED
 };
 
-/* EXTI line number for PWR Wakeup pin 1 */
-#define PWR_EXTI_WKUP1		55
-
 struct stm32_pwr_data {
 	vaddr_t base;
 	struct itr_chip *pwr_irq_chip;
@@ -154,11 +150,6 @@ static void stm32mp1_pwr_itr_enable_nolock(size_t it)
 
 	VERBOSE_PWR("Pwr irq enable");
 
-	if (IS_ENABLED(CFG_STM32_EXTI) && priv->exti) {
-		stm32_exti_unmask(priv->exti, PWR_EXTI_WKUP1 + it);
-		stm32_exti_enable_wake(priv->exti, PWR_EXTI_WKUP1 + it);
-	}
-
 	io_setbits32(priv->base + MPUWKUPENR, BIT(it));
 }
 
@@ -169,11 +160,6 @@ static void stm32mp1_pwr_itr_disable_nolock(size_t it)
 	VERBOSE_PWR("Pwr irq disable");
 
 	io_clrbits32(priv->base + MPUWKUPENR, BIT(it));
-
-	if (IS_ENABLED(CFG_STM32_EXTI) && priv->exti) {
-		stm32_exti_mask(priv->exti, PWR_EXTI_WKUP1 + it);
-		stm32_exti_disable_wake(priv->exti, PWR_EXTI_WKUP1 + it);
-	}
 }
 
 static TEE_Result stm32_pwr_irq_set_trig(size_t it, unsigned int flags)
@@ -350,10 +336,9 @@ static struct itr_chip stm32mp1_pwr_itr_chip = {
 };
 
 static TEE_Result stm32mp1_pwr_itr_dt_get(struct dt_pargs *args,
-					  void *priv_data,
+					  void *priv_data __unused,
 					  struct itr_desc *itr_desc_p)
 {
-	struct stm32_pwr_data *priv = priv_data;
 	TEE_Result res = TEE_ERROR_GENERIC;
 	unsigned int trigger_mode = 0;
 	unsigned int itr_num = 0;
@@ -412,9 +397,6 @@ static TEE_Result stm32mp1_pwr_itr_dt_get(struct dt_pargs *args,
 		return res;
 	}
 
-	if (IS_ENABLED(CFG_STM32_EXTI) && priv->exti)
-		stm32_exti_set_tz(priv->exti, PWR_EXTI_WKUP1 + itr_num);
-
 	itr_desc_p->chip = &stm32mp1_pwr_itr_chip;
 	itr_desc_p->itr_num = itr_num;
 
@@ -443,12 +425,6 @@ static TEE_Result stm32mp1_pwr_irq_probe(const void *fdt, int node,
 	if (res)
 		panic();
 
-	if (IS_ENABLED(CFG_STM32_EXTI)) {
-		res = stm32_exti_get_pdata(fdt, node, &priv->exti);
-		if (res)
-			priv->exti = NULL;
-	}
-
 	res = interrupt_dt_get(fdt, node, &itr_chip, &itr_num);
 	if (res)
 		goto err;
-- 
2.47.2

